server/athenian/api/internal/jira.py

```diff
@@ -513,18 +513,20 @@ def normalize_user_type(type_: str) -> str:
     return type_
 
 
-def parse_request_priorities(req_priorities: Optional[list[str]]) -> Optional[list[str]]:
+def parse_request_priorities(req_priorities: list[str] | None) -> list[str] | None:
     """Parse the raw Jira priorities received in a request."""
     if req_priorities is None:
         return None
-    return sorted({normalize_priority(p) for p in req_priorities})
+    normalized = [normalize_priority(p) for p in req_priorities]
+    return list({p: None for p in normalized})  # unique keeping order
 
 
-def parse_request_issue_types(req_issue_types: Optional[list[str]]) -> Optional[list[str]]:
+def parse_request_issue_types(req_issue_types: list[str] | None) -> list[str] | None:
     """Parse the raw Jira issue types received in a request."""
     if req_issue_types is None:
         return None
-    return sorted({normalize_issue_type(t) for t in req_issue_types})
+    normalized = [normalize_issue_type(t) for t in req_issue_types]
+    return list({t: None for t in normalized})
```

Comment:
```suggestion
    return list({(normalize_issue_type(t) for t in req_issue_types})
```

Reply:
set step will forget the req_issue_types order here

Reply:
```python
return list({normalize_issue_type(t): None for t in req_issue_types})
```

Reply:
otherwise, we create a list, then a dict, then a list.